cpp/src/arrow/json/options.h

```diff
@@ -53,6 +53,9 @@ struct ARROW_EXPORT ParseOptions {
   /// How JSON fields outside of explicit_schema (if given) are treated
   UnexpectedFieldBehavior unexpected_field_behavior = UnexpectedFieldBehavior::InferType;
 
+  /// Whether decimals are represented as strings(quoted) or numbers(unquoted)
+  bool parse_decimal_as_number = false;
```

Comment:
I'm wondering: instead of making this an option, would it be more expensive to support both automatically?

Reply:
> 

Ha, I think it may work. Let me experiment with it a bit.